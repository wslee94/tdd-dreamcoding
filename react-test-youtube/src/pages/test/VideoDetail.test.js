import { Route } from "react-router-dom";
import VideoDetail from "../VideoDetail";
import renderer from "react-test-renderer";
import { withRouter } from "../../tests/utils";
import { fakeVideo, fakeVideo as video } from "../../tests/videos";
import ChannelInfo from "../../components/ChannelInfo";
import RelatedVideos from "../../components/RelatedVideos";
import { render } from "@testing-library/react";

jest.mock("../../components/ChannelInfo");
jest.mock("../../components/RelatedVideos");

describe("VideoDetail", () => {
  afterEach(() => {
    ChannelInfo.mockReset();
    RelatedVideos.mockReset();
  });

  it("renders correctly", () => {
    const component = renderer.create(
      withRouter(<Route path="/" element={<VideoDetail />} />, {
        pathname: "/",
        state: { video },
      })
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("check props", () => {
    const { channelId, channelTitle } = fakeVideo.snippet;

    render(
      withRouter(<Route path="/" element={<VideoDetail />} />, {
        pathname: "/",
        state: { video },
      })
    );

    expect(ChannelInfo.mock.calls[0][0]).toStrictEqual({
      id: channelId,
      name: channelTitle,
    });

    expect(RelatedVideos.mock.calls[0][0]).toStrictEqual({
      id: video.id,
    });
  });
});
