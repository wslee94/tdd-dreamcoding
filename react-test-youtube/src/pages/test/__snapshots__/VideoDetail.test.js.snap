// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoDetail renders correctly 1`] = `
<section
  className="flex flex-col lg:flex-row"
>
  <article
    className="basis-4/6"
  >
    <iframe
      frameBorder="0"
      height="640"
      id="player"
      src="https://www.youtube.com/embed/1"
      title="title"
      type="text/html"
      width="100%"
    />
    <div
      className="p-8"
    >
      <h2
        className="text-xl font-bold"
      >
        title
      </h2>
      <pre
        className="whitespace-pre-wrap"
      />
    </div>
  </article>
  <section
    className="basis-2/6"
  />
</section>
`;
